tests/unit/s2n_server_extensions_test.c

```diff
@@ -418,7 +445,11 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_enable_tls13());
             struct s2n_connection *conn;
             EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
-            EXPECT_SUCCESS(s2n_connection_allow_all_response_extensions(conn));
+            s2n_extension_type_id extension_id = 0;
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_KEY_SHARE, &extension_id));
+            S2N_CBIT_SET(conn->extension_requests_received, extension_id);
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_SUPPORTED_VERSIONS, &extension_id));
+            S2N_CBIT_SET(conn->extension_requests_received, extension_id);
```

Comment:
It looks like the point of this test is to check the default extensions. Turning only some of them on seems like it defeats the purpose.

I'd suggest either just adding your extension to the expected size (since it is a conveniently constant size) or specifically turn your extension off.

Or, like I mentioned offline, just kill these tests :)

Reply:
I don't think this test is checking the default extensions. All of these extensions need to be explicitly turned on with a flag. For example, conn->server_name_used needs to be true before the server will send the server name extension. 
What I worry about with just adding 4 bytes to the expected size or turning off my specific extension is that this test will break in the exact same way if we add another default extension in the future. Just seems weird to me.

Reply:
Hmmm I guess that's fair. The actual name is "Test TLS13 Extensions with null key exchange alg cipher suites", so maybe it's just trying to test the key exchange extensions.

You could also maybe just set the actual protocol version to S2N_TLS13, disabling your extension, since that's what the test says its testing?